internal/outbox/gormoutbox/outbox.go

```diff
@@ -0,0 +1,57 @@
+package gormoutbox
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/vladyslavpavlenko/genesis-api-project/internal/outbox"
+
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+// Outbox is the repository for this package.
+type Outbox struct {
+	db *gorm.DB
+}
+
+// NewOutbox creates a new `events` table to serve as an outbox.
+func NewOutbox(db *gorm.DB) (*Outbox, error) {
+	err := db.AutoMigrate(&outbox.Event{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to migrate events")
+	}
+	return &Outbox{db: db}, nil
+}
+
+// AddEvent creates a new Event record.
+func (o *Outbox) AddEvent(data outbox.Data) error {
+	event := &outbox.Event{
+		Published: false,
```

Comment:
its not great idea, better have other table with stream metadata like last offset